src/View/Helper/FormHelper.php

```diff
@@ -59,24 +60,24 @@ class FormHelper extends Helper
      */
     protected $_templateSet = [
         'default' => [
-            'checkboxContainer' => '<div class="checkbox">{{content}}{{help}}</div>',
-            'checkboxContainerError' => '<div class="checkbox has-error">{{content}}{{error}}{{help}}</div>',
+            'checkboxContainer' => '<div class="form-check"{{required}}>{{content}}{{help}}</div>',
+            'checkboxContainerError' => '<div class="form-check">{{content}}{{error}}{{help}}</div>',
         ],
         'inline' => [
             'label' => '<label class="sr-only"{{attrs}}>{{text}}</label>',
             'inputContainer' => '{{content}}'
         ],
         'horizontal' => [
-            'label' => '<label class="control-label %s"{{attrs}}>{{text}}</label>',
+            'label' => '<label class="col-form-label %s"{{attrs}}>{{text}}</label>',
             'formGroup' => '{{label}}<div class="%s">{{input}}{{error}}{{help}}</div>',
             'checkboxFormGroup' => '<div class="%s"><div class="checkbox">{{label}}</div>{{error}}{{help}}</div>',
             'submitContainer' => '<div class="form-group"><div class="%s">{{content}}</div></div>',
             'inputContainer' => '<div class="form-group {{type}}{{required}}">{{content}}</div>',
-            'inputContainerError' => '<div class="form-group {{type}}{{required}} has-error">{{content}}</div>',
+            'inputContainerError' => '<div class="form-group {{type}}{{required}}">{{content}}</div>',
             'checkboxContainer' => '<div class="form-group{{required}}">{{content}}</div>',
-            'checkboxContainerError' => '<div class="form-group{{required}} has-error">{{content}}</div>',
+            'checkboxContainerError' => '<div class="form-group{{required}}">{{content}}</div>',
             'radioContainer' => '<div class="form-group{{required}}">{{content}}</div>',
-            'radioContainerError' => '<div class="form-group{{required}} has-error">{{content}}</div>',
+            'radioContainerError' => '<div class="form-group{{required}}">{{content}}</div>',
```

Comment:
Is the removal of has-error desired? Might take away some of the functionality of CSS styling in the error case?

Reply:
The Bootstrap Migration Guide mentions, that `.has-error` has been dropped in favor of `:invalid`/ `:valid` pseudo classes. See https://getbootstrap.com/docs/4.0/migration/#forms-1
For server side validation they suggest using `.is-invalid` and `.is-valid` on the wrapping element. See https://getbootstrap.com/docs/4.0/components/forms/#how-it-works